### PR TITLE
roachtest: segregate issues for coverage and metamorphic builds

### PR DESCRIPTION
**Pull request by RaduBerinde**
This PR is now just for the top commit. The rest is #115302.

#### roachtest: segregate issues for coverage and metamorphic builds

Issues posted for metamorphic/coverage builds now have
`B-metamorphic`/`B-coverage` labels and no `release-blocker` label.
We also include top-level notices in the issue description with
information about these builds.

We adopt an existing issue only when the metamorphic/coverage labels
match exactly.

Informs: #114615
Release note: None

### CODE REVIEW

**Review by renatolabs — APPROVED** [review]
:lgtm:

Small nits and a question about release-blocker. We should probably wait until next week before we merge this one.

> Reviewed 2 of 2 files at r1, 31 of 31 files at r2, 11 of 11 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-:-Njt35PPDsb29sTWNbEm:blzyq8a)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @DarrylWong, @herkolategan, @RaduBerinde, and @srosenberg)

___
*[`pkg/cmd/roachtest/github.go` line 183 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-Njt2CJB4Qscfi6i_GBI:-Njt2CJB4Qscfi6i_GBJ:b-v2cw1n) ([raw file](https://github.com/cockroachdb/cockroach/blob/933d5b42decb2b3b92849edc799dac714f7eda3d/pkg/cmd/roachtest/github.go#L183)):*
> ```Go
> 	if infraFlake {
> 		labels = append(labels, "X-infra-flake")
> 	} else if !spec.NonReleaseBlocker && !metamorphicBuild && !coverageBuild {
> ```

This is a little extreme, IMO? Maybe while we stabilize these features it's fine, but in the long run I don't think they should be treated differently.
___
*[`pkg/cmd/roachtest/github.go` line 264 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-Njt2W-hDpHu7JiH76tg:-Njt2W-hDpHu7JiH76th:b-q0i5wa) ([raw file](https://github.com/cockroachdb/cockroach/blob/933d5b42decb2b3b92849edc799dac714f7eda3d/pkg/cmd/roachtest/github.go#L264)):*
> ```Go
> 			"This is a special code-coverage build. If the same failure was hit in a non-coverage run, "+
> 				"there should be a similar issue without the B-coverage label. If not, it is possible that "+
> 				"this failure is related to the code coverage infrastructure or overhead")
> ```

Nits: use `coverageLabel` const (same for message below); missing `.` at end of last sentence.
___
*[`pkg/cmd/roachtest/github.go` line 270 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-Njt2nbE2hXPKlpzo-S3:-Njt2nbF4Xh27VBGYdjx:b-w8aloy) ([raw file](https://github.com/cockroachdb/cockroach/blob/933d5b42decb2b3b92849edc799dac714f7eda3d/pkg/cmd/roachtest/github.go#L270)):*
> ```Go
> 			"This build has metamorphic test constants enabled. If the same failure was hit in a "+
> 				"non-metamorphic run, there should be a similar issue without the B-metamorphic label.If not, it is possible "+
> 				"that this failure is caused by a metamorphic constant")
> ```

Nits: missing space in `label.If not`; missing `.` at the end of last sentence.

**Review by RaduBerinde — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-:-NkHJyyc5GsvyMJwpgB-:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @DarrylWong, @herkolategan, @renatolabs, and @srosenberg)

___
*[`pkg/cmd/roachtest/github.go` line 183 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-Njt2CJB4Qscfi6i_GBI:-NkHJc6w2YYEDkRqHFn0:b1utpzl) ([raw file](https://github.com/cockroachdb/cockroach/blob/933d5b42decb2b3b92849edc799dac714f7eda3d/pkg/cmd/roachtest/github.go#L183)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

This is a little extreme, IMO? Maybe while we stabilize these features it's fine, but in the long run I don't think they should be treated differently.
</blockquote></details>

I added a TODO to remove the check once they are stable.
___
*[`pkg/cmd/roachtest/github.go` line 264 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-Njt2W-hDpHu7JiH76tg:-NkHJqj_6obEUtakAjzf:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/933d5b42decb2b3b92849edc799dac714f7eda3d/pkg/cmd/roachtest/github.go#L264)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Nits: use `coverageLabel` const (same for message below); missing `.` at end of last sentence.
</blockquote></details>

Done.
___
*[`pkg/cmd/roachtest/github.go` line 270 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-Njt2nbE2hXPKlpzo-S3:-NkHJv1X1invPoqZoQTG:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/933d5b42decb2b3b92849edc799dac714f7eda3d/pkg/cmd/roachtest/github.go#L270)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Nits: missing space in `label.If not`; missing `.` at the end of last sentence.
</blockquote></details>

Done.

**Comment by RaduBerinde** [comment]
I started a coverage run on this branch and will look at any resulting issues.

**Review by srosenberg — APPROVED** [review]
:lgtm:  Thanks!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-:-NkHmqhC8pEJHfaI6SSs:b-e7bejd)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @DarrylWong, @herkolategan, @RaduBerinde, and @renatolabs)

___
*[`pkg/cmd/internal/issues/issues.go` line 484 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-NkHmlWG8O3wCJpFMITp:-NkHmlWG8O3wCJpFMITq:b-ze7n6e) ([raw file](https://github.com/cockroachdb/cockroach/blob/1513c7d090c93eaf859f81cf3eee0865d7d17fc5/pkg/cmd/internal/issues/issues.go#L484)):*
> ```Go
> 	// TopLevelNotes are messages that are printed prominently at the top of the
> 	// issue description.
> 	TopLevelNotes []string
> ```

How do you envision these?

**Review by RaduBerinde — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114625)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 2 stale) (waiting on @DarrylWong, @herkolategan, @renatolabs, and @srosenberg)

___
*[`pkg/cmd/internal/issues/issues.go` line 484 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/114625#-NkHmlWG8O3wCJpFMITp:-NkHpepe4wHAIoqVOHU2:b4jn67k) ([raw file](https://github.com/cockroachdb/cockroach/blob/1513c7d090c93eaf859f81cf3eee0865d7d17fc5/pkg/cmd/internal/issues/issues.go#L484)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

How do you envision these?
</blockquote></details>

These are just prepended to the issue description with a `**Note:**`. You can see the notes I added in `github.go` for metamorphic and coverage builds (I had forgotten to set `TopLevelNotes`, fixed now).

**Comment by RaduBerinde** [comment]
> I started a coverage run on this branch and will look at any resulting issues.

That won't work, it doesn't post from a PR branch: `skipping GitHub issue posting (not a release branch: "114625")`

**Comment by srosenberg** [comment]
> > I started a coverage run on this branch and will look at any resulting issues.
> 
> That won't work, it doesn't post from a PR branch: `skipping GitHub issue posting (not a release branch: "114625")`

Oops, I meant to say that but figured you were probably just looking to test for unrelated failures... There is `TestPostEndToEnd` which can be readily used for manually testing the issue poster.

**Comment by RaduBerinde** [comment]
I got a test run to file an issue: https://github.com/cockroachdb/cockroach/issues/115236
It looks good, other than still having the `release-blocker` label, I will investigate.

**Comment by renatolabs** [comment]
> currently does nothing?

Yep, I think so. ☹️ 

> adds the release-blocker label whenever we are adding the test-failure label

IIRC this change was made in response to Code Yellow's policy of treating every test failure as a release blocker by default. I think we still want that behaviour, but we'll need a way to tell the issue poster to allow only `C-test-failure` for cases like these.

**Comment by RaduBerinde** [comment]
I am spinning out the various `issues` fixes and improvements in #115302. I will rebase and adjust this PR once that is green.

**Comment by RaduBerinde** [comment]
This (and base #115302) is ready to go. Once it's in, I will reenable coverage runs twice a week.

**Comment by RaduBerinde** [comment]
Friendly ping.

**Review by srosenberg — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 34 of 34 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114625)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 2 stale) (waiting on @DarrylWong, @herkolategan, and @renatolabs)

**Pull request merged**
